### ISSUE TITLE
Upload fails for large files

### ISSUE THREAD

**Issue by olivercalder** [nextcloud-snap/nextcloud-snap#2200]
#### Describe the bug
Uploads of large files are failing on my install. According to the Nextcloud documentation ([https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/big_file_upload_configuration.html](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/big_file_upload_configuration.html)), several adjustments to the configuration are required to allow for large files. Namely, adjustments to PHP and Apache.

For PHP, the one must modify the upload_max_filesize and post_max_filesize values. However, I have found these in two places, and it is unclear which should be modified:

`/snap/nextcloud/current/config/php/php.ini`:

```
post_max_size = 8M
upload_max_filesize = 2M
```

`/snap/nextcloud/current/config/php/php-fpm.d/www.conf`:

```
php_value[upload_max_filesize] = 16G
php_value[post_max_size] = 16G
```

Additionally, assuming changing these values would solve the issue, is there a way to make these changes persist across updates to the nextcloud-snap package? The same goes for modifications to `/var/snap/nextcloud/current/nextcloud/config/config.php` via `nextcloud.occ` or manually.

#### To Reproduce
1. Install Nextcloud
2. Set up the desktop client to sync to a local folder
3. Move a very large file (>20G) into the folder
4. See error in desktop client

#### OS/snapd/snap version
I am running Raspberry Pi OS based on Debian Bullseye:

```
$ snap list nextcloud
Name       Version      Rev    Tracking       Publisher   Notes
nextcloud  24.0.6snap1  31759  latest/stable  nextcloud✓  -
```

```
$ snap --version
snap    2.57.2
snapd   2.57.2
series  16
debian  11
kernel  5.15.61-v8+
```

#### Logs
Logs are huge, I was manually editing `/var/snap/nextcloud/current/nextcloud/config/config.php` and my nginx config (I'm running a reverse proxy on the same machine) and I messed up the syntax for a while, so I kept restarting nextcloud and nginx. Logs are attached nonetheless:
[output.log](https://github.com/nextcloud-snap/nextcloud-snap/files/9740962/output.log)

**Comment by kyrofa** [comment]
> 2. Set up the desktop client to sync to a local folder
> 3. Move a very large file (>20G) into the folder
> 4. See error in desktop client

The desktop client should be chunking, and thus never approach that limit.

**Comment by olivercalder** [comment]
The error I was getting was "413 Request entity too large" on a PUT request. I thought this might be from my nginx reverse proxy, so I tried adding `client_max_body_size = 4M;`, as well as `16M`, `2048M`, and `16G`, and none of those adjustments solved the issue. Most of the sites I'm seeing online discussing this issue also suggest modifying `upload_max_filesize` and `post_max_size` in `php.ini`. However, `php.ini` is read only in the snap mounted filesystem. I'm not sure what else to try. Do you have any suggestions?

**Comment by szaimen** [comment]
Looking at the snaps config, may the Apache and Apache mod_proxy_fcgi not be correct for big uploads based on https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/big_file_upload_configuration.html#apache and below

**Comment by kyrofa** [comment]
@olivercalder is saying it happens for files as small as 4MB though, @szaimen. @olivercalder this may indeed be caused by nginx. Have you taken a look at its logs? Do you see similar behavior through the web interface?

**Comment by szaimen** [comment]
> @olivercalder is saying it happens for files as small as 4MB though, @szaimen. @olivercalder this may indeed be caused by nginx. Have you taken a look at its logs? Do you see similar behavior through the web interface?

ah I see. That shouldn't happen for such small files, indeed

**Comment by olivercalder** [comment]
I left this for a few days and now after turning on sync again, I'm not seeing the issue anymore. I didn't change anything at all since then, so this is rather odd. I did check the nginx logs, and you're absolutely right, there are various 413 errors from when I previously tried uploading, but again, the problem seems to have resolved itself since then. Notably, I did set `client_max_body_size = 2048m` in both `nginx.conf` and the reverse proxy config in `sites-enabled`, so that may well have been the problem (and it needed some time to settle despite my restarting nginx, nextcloud, and the desktop client several times ??). Regardless, thank you both for your time and suggestions. I really appreciate the work you do!
